src/Diagnostics.ModelsAndUtils/Models/OperationContext.cs

```diff
@@ -7,34 +7,48 @@ namespace Diagnostics.ModelsAndUtils.Models
     /// <summary>
     /// Operation Context
     /// </summary>
-    public class OperationContext
+    public class OperationContext<TResource> where TResource : IResource
     {
         /// <summary>
-        /// Site Resource
+        /// Resource Object
+        /// See <see cref="App"/> and <see cref="HostingEnvironment"/>
         /// </summary>
-        public SiteResource Resource;
+        public TResource Resource { get; private set; }
 
         /// <summary>
         /// Start Time(UTC) for data measurement
         /// </summary>
-        public string StartTime;
+        public string StartTime { get; private set; }
```

Comment:
Should we do the work of turning the start and end time string into an actual datetime before passing it to the csx? I guess if the helper methods use string then it makes sense, but it seems like if they need to parse the start or end time in csx then it could lead to people doing it wrong.

Reply:
I made another comment about this in a more relevant place below so look for that one

Reply:
Good point. I will include that. That will lead to few changes down the pipe till data providers, but i guess right time to do this change